File: BibliotecaVirtual/Editora.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BibliotecaVirtual
{
    class Editora
    {
        private string nomeSolicitante { get; set; }
        private string endereco { get; set; }
        private string email { get; set; }

        public string GetEmail()
        {
            return email;
        }
        public void SetEmail(string email)
        {
            this.email = VerificaEmail(email);
        }
        public string GetEndereco()
        {
            return endereco;
        }
        public void SetEndereco(string endereco)
        {
            this.endereco = endereco;
        }
        public string GetNome()
        {
            return nomeSolicitante;
        }
        public void SetNome(string nome)
        {
            this.nomeSolicitante = nome;
        }

        public void lerEditora()
        {
            Console.WriteLine("Digite o nome da editora: ");
            Console.ReadLine();

            Console.WriteLine("Digite o endereço da editora: ");
            Console.ReadLine();

            Console.WriteLine("Digite o email da editora: ");
            Console.ReadLine();
        }


        public string VerificaEmail(string email)
        {
            bool x;
            char a;

            do
            {
                x = true;

                for (int i = 0; i < email.Length; i++)
                {
                    a = email[i];
                    int b = a;
                    if (b == 64)
                    {
                        x = false;
                    }
                }
                if (x)
                {
                    Console.WriteLine("Email inválido\n Informe um email válido");
                    Console.ReadLine();
                }
            } while (x);
            return email;
        }

        public string toString()
        {
            return "" + nomeSolicitante + "";
        }
    }
}

File: Biblioteca_Java/Editora.java
package Biblioteca;

import java.util.Scanner;

public class Editora {
    private String nomeSolicitante;
    private String endereco;
    private String email;

    Scanner teclado = new Scanner(System.in);

    public void lerEditora(){
        System.out.println("Digite o nome da editora: ");
        this.setNome(teclado.nextLine());

        System.out.println("Digite o endereço da editora: ");
        this.setEndereco(teclado.nextLine());

        System.out.println("Digite o email da editora: ");
        this.setEmail(teclado.nextLine());

    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = verificaEmail(email);
    }

    public String getEndereco() {
        return endereco;
    }

    public void setEndereco(String endereco) {
        this.endereco = endereco;
    }

    public String getNome() {
        return nomeSolicitante;
    }

    public void setNome(String nome) {
        this.nomeSolicitante = nome;
    }

    public String verificaEmail(String email){
        boolean x;
        char a;

        do {
            x = true;

            for (int i = 0; i < email.length(); i++){
                a = email.charAt(i);
                int b = a;
                if(b == 64){
                    x = false;
                }
            }
            if(x){
                System.out.println("Email inválido\nInforme um email válido");
                email = teclado.nextLine();
            }
        }while(x);
        return email;
    }

    @Override
    public String toString() {
        return "" + nomeSolicitante + "";
    }
}
